python/cuml/tests/test_umap.py

```diff
@@ -925,3 +926,49 @@ def test_umap_small_fit_large_transform():
 
     trust = trustworthiness(infer, embeddings, n_neighbors=10)
     assert trust >= 0.9
+
+
+def test_umap_outliers():
```

Comment:
It is not immediately clear to me how this test is testing specifically for outliers. I also think we should probably parametrize this test as described [here](https://github.com/rapidsai/cuml/blob/branch-25.10/wiki/python/DEVELOPER_GUIDE.md#testing-and-unit-testing) for improved test coverage.

Reply:
we test to see if there are values in the final embedding that are way out of range compared to the cpu umap output.
in small datasets (I didn't want to test with anything too large here because that takes a long time for cpu umap to run) outliers don't happen that often, so it is difficult to parameterize the test based on the #rows and #features of the dataset. This dataset was specifically chosen because I've verified that it produces outliers without the fix in this PR.

We can parametrize umap related params though.

Reply:
I'd recommend adding a few code comments that make the approach here more obvious.

Parametrizing on uamp params is probably a good idea. I'll let you judge what would be appropriate.